packages/apollos-ui-auth/src/AuthSMS/PhoneEntry.js

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable react/no-unused-prop-types */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { KeyboardAvoidingView, ScrollView, StyleSheet } from 'react-native';
+import { SafeAreaView } from 'react-navigation';
+import {
+  H6,
+  PaddedView,
+  BackgroundView,
+  TextInput,
+  FlexedView,
+  ButtonLink,
+} from '@apollosproject/ui-kit';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import gql from 'graphql-tag';
+import { Mutation } from 'react-apollo';
+
+import {
+  NextButtonRow,
+  NextButton,
+  TitleText,
+  PromptText,
+  BrandIcon,
+} from '../styles';
+
+const requestPin = gql`
+  mutation requestPin($phone: String!) {
+    requestSmsLoginPin(phoneNumber: $phone) {
+      success
+    }
+  }
+`;
+
+class PhoneEntry extends Component {
+  static propTypes = {
+    brand: PropTypes.node,
+    authTitleText: PropTypes.string,
+    smsPromptText: PropTypes.string,
+    smsPolicyInfo: PropTypes.node,
+    allowPassword: PropTypes.bool,
+    smsPasswordLoginPrompt: PropTypes.node,
+    screenProps: PropTypes.shape({}), // we'll funnel screenProps into props
+  };
+
+  static defaultProps = {
+    brand: <BrandIcon />,
+    authTitleText: 'Have we met before?',
+    smsPromptText:
+      "Lets get you signed in using your mobile number. We'll text you a code to make login super easy!",
+    smsPolicyInfo: (
+      <H6>
+        {"We'll never share your information or contact you (unless you ask!)."}
+      </H6>
+    ),
+    allowPassword: true,
+    smsPasswordLoginPrompt: "I'd rather use my email and a password",
+  };
+
+  validationSchema = Yup.object().shape({
+    phone: Yup.string().matches(/^\D?(\d{3})\D?\D?(\d{3})\D?(\d{4})$/),
+  });
+
+  get flatProps() {
+    return { ...this.props, ...(this.props.screenProps || {}) };
+  }
+
+  submitHandler = (mutate) => async (
```

Comment:
I'm surprised our linter didn't pick this up but typically we try to enforce something like this.

```suggestion
  handleOnSubmit = (mutate) => async (
```

Reply:
`handleOnSubmit` is incorrect. This is a function that returns a function, so it is named differently (and should be named differently). I'm happy to consder a new name to make this more clear